### PR TITLE
Visual Studio 2017 compatibility fix

### PR DESCRIPTION
**Pull request by TDannhauer**
Some minor changes to make ninja compilable with MS Visual Studio 2017

### CODE REVIEW

File: src/clparser.cc [hunk 1]
```diff
@@ -53,15 +53,15 @@ string CLParser::FilterShowIncludes(const string& line,
 
 // static
 bool CLParser::IsSystemInclude(string path) {
-  transform(path.begin(), path.end(), path.begin(), ::tolower);
+  transform(path.begin(), path.end(), path.begin(), [](const char c) { return static_cast<char>(::tolower(c)); });
```
Comment (nico) [inline src/clparser.cc]: Ninja doesn't use any c++11 features and doesn't pass -std=c++11 to gcc/clang (where one has to explicitly opt in to that).

Comment (TDannhauer) [inline src/clparser.cc]: MS VC compiler has afaik no switches to select specific c++ versions. Regardless which c++ version is used in ninja, it is not compilable with Visual Studio 2017 and needs to be fixed. 

Any suggestions?

Comment (paulsapps) [inline src/clparser.cc]: MS VC actually does have switches now https://blogs.msdn.microsoft.com/vcblog/2016/06/07/standards-version-switches-in-the-compiler/

"We’re creating a series of new switches that correspond to versions of the C++ language, starting with /std:c++14 and /std:c++latest."

**Comment by TDannhauer** [comment]
@paulsapps thanks. Does this also apply to VS 2017? They will provides only switches for C++14 and later.

Someone else if the ninja team hast to chip in to make it compatible with VS 2017. I'll quit now since my time is limited and ninja not my focus.

Thanks so far and good luck.

**Comment by paulsapps** [comment]
@TDannhauer it applies to 2015+

**Comment by jhasse** [comment]
AppVeyor uses VS2017 and it seems to work fine without this. Closing.

**Pull request closed without merging**